docroot/modules/custom/civictheme_migrate/src/Form/CivicThemeMigrationConfigurationForm.php

```diff
@@ -0,0 +1,500 @@
+        'file_validate_extensions' => ['json txt'],
+        'civictheme_migrate_validate_json' => [],
+      ],
+    ];
+
+    $form['actions']['#type'] = 'actions';
+    $form['actions']['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->getSaveConfigurationLabel(),
+      '#button_type' => 'primary',
+    ];
+    $form['actions']['generate_migration'] = [
+      '#type' => 'submit',
+      '#value' => $this->getGenerateMigrationLabel(),
+      '#button_type' => 'primary',
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    if ($this->isGenerateMigrationSubmit($form_state)) {
+      $files = $form_state->getValue('configuration_files');
+      if (empty($files)) {
+        $form_state->setError($form['configuration_files'], $this->t('Configuration files are required to generate a migration'));
+      }
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $config = $this->config('civictheme_migrate.settings');
+    if ($this->isRetrieveFilesSubmit($form_state)) {
+      $urls = explode("\r\n", $form_state->getValue('endpoint'));
+      try {
+        $files = $this->retrieveFiles($urls);
+        $existing_files = $config->get('configuration_files') ?? [];
+        $config->set('configuration_files', array_merge($existing_files, $files));
+        $this->messenger()->addStatus($this->t('Merlin extracted content JSON files have been retrieved'));
+      }
+      catch (\Exception $exception) {
+        $this->messenger()->addError($exception->getMessage());
+      }
+    }
+
+    if ($this->isSaveConfigurationSubmit($form_state) || $this->isGenerateMigrationSubmit($form_state)) {
+      $config->set('configuration_files', $form_state->getValue('configuration_files'));
+    }
+    $config->set('migration_type', $form_state->getValue('migration_type'));
+    $config->set('remote', [
+      'auth_type' => $form_state->getValue('auth_type'),
+      'auth_username' => $form_state->getValue('auth_username'),
+      'auth_password' => $form_state->getValue('auth_password'),
+      'auth_token' => $form_state->getValue('auth_token'),
+      'endpoint' => $form_state->getValue('endpoint'),
+    ]);
+    $config->save();
+    $this->messenger()->addStatus($this->t('The configuration options have been saved.'));
+
+    if ($this->isGenerateMigrationSubmit($form_state)) {
+      $this->generateMigrationConfigurationSubmit($form, $form_state);
+    }
+
+  }
+
+  /**
+   * Handles submission to generate and import migration configuration.
+   *
+   * @param array $form
+   *   Form array.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   Formstate object.
+   */
+  protected function generateMigrationConfigurationSubmit(array &$form, FormStateInterface $form_state) {
+    $file_ids = $form_state->getValue('configuration_files');
```

Comment:
can all of this be moved to a MigrationManager class and handled there. The config form class (this class) is only responsible for rendering form and dispatching submissions